pandas_datareader/oandarest.py

```diff
@@ -0,0 +1,376 @@
+                    M  ->  1 month candlesticks, aligned to first day of the month
+                    See OANDA REST v20 for updated list
+            Default: See DEFAULT_FREQUENCY
+        candleFormat: string
+            Candlesticks representations. Eg: BA to get bid and ask values
+            Valid values:
+                M for midpoint
+                B for Bid
+                A for Ask
+            Default: See DEFAULT_CANDLE_FORMAT
+        access_credential: Dict of strings
+            Credential to query the api
+            credential["accountType"]="practise". Mandatory. Valid values: practice, live
+            credential["apiToken"]="Your OANDA API token". Mandatory. Valid value: See your OANDA Account's API Token
+
+        Returns:
+        Panel with currency pairs requested. Each dataframe represents a currency pair and is MultiIndexed enabled.
+        Each dataframe data point can be addressed as is (assuming Midpoint data were requested):
+            df['Mid']['Open']
+            df['Mid']['High']
+            df['Mid']['Low']
+            df['Mid']['Close']
+            df['Mid']['Volume']
+            df['Mid']['Complete']
+    """
+
+    DEFAULT_FREQUENCY = "5S"
+    DEFAULT_CANDLE_FORMAT = "M"
+    SUPPORTED_OFFSET_ALIASES = {
+        "5S": "S5",
+        "10S": "S10",
+        "15S": "S15",
+        "30S": "S30",
+        "T": "M1",
+        "2T": "M2",
+        "4T": "M4",
+        "5T": "M5",
+        "15T": "M15",
+        "30T": "M30",
+        "H": "H1",
+        "2H": "H2",
+        "3H": "H3",
+        "4H": "H4",
+        "6H": "H6",
+        "8H": "H8",
+        "12H": "H12",
+        "1M": "M"
+    }
+
+    def __init__(self, symbols, symbolsTypes=None,
+                 start=None, end=None,
+                 freq=None, candleFormat=None,
+                 session=None,
+                 access_credential=None):
+        _BaseReader.__init__(self, symbols, start=start,
+                             end=end, session=session)
+
+        self.symbols = symbols
+        if symbols is None:
+            self.symbols = ["EUR_USD"]
+
+        if type(symbols) is str:
+            self.symbols = [symbols]
+
+        self.symbolsTypes = symbolsTypes
+        if symbolsTypes is None:
+            self.symbolsTypes = ["currency"]
+
+        if len(self.symbols) != len(self.symbolsTypes):
+            self.symbolsTypes = ["currency" for x in self.symbols]
+
+        # print(self.symbols)
+        # print(self.symbolsTypes)
+
+        self.freq = freq
+        if freq is None:
+            self.freq = OANDARestHistoricalInstrumentReader.DEFAULT_FREQUENCY
+
+        self.candleFormat = candleFormat
+        if candleFormat is None:
+            self.candleFormat = OANDARestHistoricalInstrumentReader.DEFAULT_CANDLE_FORMAT
+
+        self.access_credential = access_credential
+        if access_credential is None:
+            self.access_credential = {}
+
+        if 'accountType' not in access_credential:
```

Comment:
Here it's problematic

So

    if 'accountType' not in access_credential:

should be

    if 'accountType' not in self.access_credential:

because

```python
pdr.DataReader("EUR_USD", "oanda_historical_currency", "2014-03-19T09:00:00Z", "2014-03-20T09:00:00Z")
```

raises

    TypeError: argument of type 'NoneType' is not iterable